### PR TITLE
Add xtr 1.0.0

### PR DESCRIPTION
**Pull request by choll**
Specify library name and version:  **xtr/1.0.0**

This is a new library, I am the author.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/xtr/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        "fPIC": [True, False],
+        "enable_exceptions": [True, False],
+        "enable_lto": [True, False]}
+    default_options = {
+        "fPIC": True,
+        "enable_exceptions": True,
+        "enable_lto": False}
+    generators = "make"
+
+    def source(self):
```
Comment (madebr) [inline recipes/xtr/all/conanfile.py:32]:
```suggestion
    def validate(self):
        if self.settings.os not in ("FreeBSD", "Linux"):
            raise ConanInvalidConfiguration(f"Unsupported os={self.settings.os}")
        if self.settings.compiler not in ("gcc", "clang"):
            raise ConanInvalidConfiguration(f"Unsupported compiler={self.settings.compiler}")
        if self.settings.arch not in ("x86_64", ):
            raise ConanInvalidConfiguration(f"Unsupported arch={self.settings.arch}")

        minimal_cpp_standard = 20
        if self.settings.compiler.cppstd:
            tools.check_min_cppstd(self, minimal_cpp_standard)

        minimum_version = {"gcc": 10, "clang": 12}
        compiler = str(self.settings.compiler)
        version = tools.Version(self.settings.compiler.version)

        if version < minimum_version[compiler]:
            raise ConanInvalidConfiguration(
                f"{self.name} requires {self.settings.compiler} version {minimum_version[compiler]} or later")

    def source(self):
```

File: recipes/xtr/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        minimum_version = {"gcc": 10, "clang": 12}
+        compiler = str(self.settings.compiler)
+        version = tools.Version(self.settings.compiler.version)
+
+        if version < minimum_version[compiler]:
+            raise ConanInvalidConfiguration(
+                "%s requires %s version %d or later"
+                % (self.name, compiler, minimum_version[compiler]))
+
+    def build(self):
```
Comment (madebr) [inline recipes/xtr/all/conanfile.py:49]: Moved to `validate`.
```suggestion
        tools.get(**self.conan_data["sources"][self.version], strip_root=True)

    def build(self):
```

File: recipes/xtr/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            str(int(bool(self.options.enable_exceptions)))
+        env_build_vars["LTO"] = str(int(bool(self.options.enable_lto)))
+        autotools.make(vars=env_build_vars)
+        autotools.make(vars=env_build_vars, target="xtrctl")
+
+    def package(self):
+        self.copy("*.hpp", dst="include", src="include")
+        self.copy("*/libxtr.a", dst="lib", src="build", keep_path=False)
+        self.copy("*/xtrctl", dst="bin", src="build", keep_path=False)
+        self.copy("LICENSE", "licenses")
```
Comment (madebr) [inline recipes/xtr/all/conanfile.py:67]: We try to keep the packages as lean as possible.
This also means removing all documentation.
```suggestion
        self.copy("LICENSE", dst="licenses")
        self.copy("*.hpp", src="include", dst="include")
        self.copy("*/libxtr.a", src="build", dst="lib", keep_path=False)
        self.copy("*/xtrctl", src="build", dst="bin", keep_path=False)

        tools.rmdir(os.path.join(self.package_folder, "man"))
```

File: recipes/xtr/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+    def package_info(self):
+        self.cpp_info.libs = ["xtr"]
+        self.cpp_info.system_libs = ["pthread"]
+        self.env_info.PATH.append(os.path.join(self.package_folder, "bin"))
+        self.env_info.MANPATH.append(os.path.join(self.package_folder, "man"))
```
Comment (madebr) [inline recipes/xtr/all/conanfile.py:73]:
```suggestion
        bin_path = os.path.join(self.package_folder, "bin")
        self.output.info(f"Appending PATH environment variable: {bin_path}")
        self.env_info.PATH.append(bin_path)
```

File: recipes/xtr/all/test_package/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.1)
+project(PackageTest CXX)
+
+set(CMAKE_CXX_STANDARD 20)
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_executable(example example.cpp)
+target_link_libraries(example ${CONAN_LIBS})
```
Comment (madebr) [inline recipes/xtr/all/test_package/CMakeLists.txt:9]: As part of going to conan 2.0, the use of the `cmake` generator to provide variables (in this case `${CONAN_LIBS}`) is phased out.
So use targets instead.
```suggestion
conan_basic_setup(TARGETS)

find_package(xtr REQUIRED)

add_executable(${PROJECT_NAME} example.cpp)
target_link_libraries(${PROJECT_NAME} PRIVATE xtr::xtr)
```

File: recipes/xtr/all/test_package/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+        cmake = CMake(self)
+        # Current dir is "test_package/build/<build_id>" and CMakeLists.txt is
+        # in "test_package"
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self):
+            os.chdir("bin")
+            self.run(".%sexample" % os.sep)
```
Comment (madebr) [inline recipes/xtr/all/test_package/conanfile.py:20]:
```suggestion
from conans import ConanFile, CMake, tools
import os


class TestPackageConan(ConanFile):
    settings = "os", "arch", "compiler", "build_type"
    generators = "cmake", "cmake_find_package_multi"

    def build(self):
        cmake = CMake(self)
        cmake.configure()
        cmake.build()

    def test(self):
        if not tools.cross_building(self):
            bin_path = os.path.join("bin", "test_package")
            self.run(bin_path, run_environment=True)
```

File: recipes/xtr/all/test_package/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.1)
+project(PackageTest CXX)
```
Comment (madebr) [inline recipes/xtr/all/test_package/CMakeLists.txt:2]:
```suggestion
project(test_package CXX)
```

**Review by madebr — COMMENTED** [review]
Thanks for contributing a recipe for your library!

File: recipes/xtr/all/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,80 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+
+import os
+
+
+class XtrConan(ConanFile):
+    name = "xtr"
```
Comment (uilianries) [inline recipes/xtr/all/conanfile.py:8]: I found other 2 projects using the very same name:

https://github.com/hypersoft/xtr
https://github.com/KacperKuryllo/xtr

I didn't check Bitbucket and Gitlab, but I'm fine accepting your package name as xtr only, because it's active (others seems be dead).

File: recipes/xtr/all/conanfile.py [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+        if version < minimum_version[compiler]:
+            raise ConanInvalidConfiguration(
+                f"{self.name} requires {self.settings.compiler} version {minimum_version[compiler]} or later")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True)
+
+    def build(self):
+        autotools = AutoToolsBuildEnvironment(self)
```
Comment (uilianries) [inline recipes/xtr/all/conanfile.py:54]: Have you considered Cmake support? Just curious.

File: recipes/xtr/all/conanfile.py [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+
+class XtrConan(ConanFile):
+    name = "xtr"
+    description = \
+        "C++ Logging Library for Low-latency or Real-time Environments"
+    topics = ("xtr", "logging", "logger")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/choll/xtr"
+    # 8.0.0 is excluded due to https://github.com/fmtlib/fmt/issues/2377
+    requires = "fmt/[>=6.0.0 < 8.0.0 || > 8.0.0]"
```
[the remark this answers is not in the document]
Comment (choll) [reply recipes/xtr/all/conanfile.py]: I've added this---thank you very much for all the code provided in the suggested fixes, it is appreciated.

In the requirements method I've used a fixed version, e.g:

```
    def requirements(self):
        self.requires("fmt/7.1.3")
```

Is this correct? Thanks

File: recipes/xtr/all/conanfile.py (hunk 9 above)
Comment (choll) [reply recipes/xtr/all/conanfile.py:61]: Yes, I'd like to add CMake support in the future. I just need to learn it first :-)

File: recipes/xtr/all/conanfile.py (hunk 8 above)
Comment (choll) [reply recipes/xtr/all/conanfile.py:8]: Thanks

File: recipes/xtr/all/conanfile.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (madebr) [reply recipes/xtr/all/conanfile.py]: 💯 % correct.

**Review by danimtb — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

File: recipes/xtr/all/conanfile.py (hunk 8 above)
Comment (prince-chrismc) [reply recipes/xtr/all/conanfile.py:8]: https://repology.org/projects/x/?search=xtr 🤔 kinda maybe used? but no longer active?
Only because logging library are so common I'd prefer to see the author name

File: recipes/xtr/all/conanfile.py [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+        "enable_lto": False,
+    }
+    generators = "make"
+
+    def requirements(self):
+        self.requires("fmt/7.1.3")
+
+    def validate(self):
+        if self.settings.os not in ("FreeBSD", "Linux"):
+            raise ConanInvalidConfiguration(f"Unsupported os={self.settings.os}")
```
Comment (prince-chrismc) [inline recipes/xtr/all/conanfile.py:33]: To date we have not permitted f-strings [there's 6 commits removing them in PRs](https://github.com/conan-io/conan-center-index/search?q=f-string&type=commits)... I see we have to approvals, are we allowing this now?

File: recipes/xtr/all/conanfile.py (hunk 8 above)
Comment (choll) [reply recipes/xtr/all/conanfile.py:8]: Do you mean adding `author = "..."` ?

File: recipes/xtr/all/conanfile.py (hunk 11 above)
Comment (SSE4) [reply recipes/xtr/all/conanfile.py:33]: so-called f-strings are Python 3.6+ : https://www.python.org/dev/peps/pep-0498/
our policy about python versions: https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#what-is-the-policy-for-supported-python-versions
TLDR: CCI supports 3.6+, there is no reason to disallow this particular feature in recipes

**Review by SSE4 — APPROVED** [review]

File: recipes/xtr/all/conanfile.py (hunk 8 above)
Comment (uilianries) [reply recipes/xtr/all/conanfile.py:8]: Sorry, I mean the project's author as package name prefix: `name = "choll-xtr"`
